src/parameterizations/vertical/MOM_ALE_sponge.F90

```diff
@@ -830,188 +903,266 @@ subroutine apply_ALE_sponge(h, dt, G, GV, US, CS, Time)
+                     spongeOnGrid=CS%SpongeDataOngrid, m_to_Z=US%m_to_Z, &
+                     answers_2018=CS%hor_regrid_answers_2018)
+      allocate( hsrc(nz_data) )
+      allocate( tmpT1d(nz_data) )
+      do c=1,CS%num_col
+        i = CS%col_i(c) ; j = CS%col_j(c)
+        CS%Ref_val(m)%p(1:nz_data,c) = sp_val(i,j,1:nz_data)
+        ! Build the source grid
+        zTopOfCell = 0. ; zBottomOfCell = 0. ; nPoints = 0; hsrc(:) = 0.0; tmpT1d(:) = -99.9
+        do k=1,nz_data
+          if (mask_z(CS%col_i(c),CS%col_j(c),k) == 1.0) then
+            zBottomOfCell = -min( z_edges_in(k+1), G%bathyT(CS%col_i(c),CS%col_j(c)) )
+            tmpT1d(k) = sp_val(CS%col_i(c),CS%col_j(c),k)
+          elseif (k>1) then
+            zBottomOfCell = -G%bathyT(CS%col_i(c),CS%col_j(c))
+            tmpT1d(k) = tmpT1d(k-1)
+          else ! This next block should only ever be reached over land
+            tmpT1d(k) = -99.9
+          endif
+          hsrc(k) = zTopOfCell - zBottomOfCell
+          if (hsrc(k)>0.) nPoints = nPoints + 1
+          zTopOfCell = zBottomOfCell ! Bottom becomes top for next value of k
+        enddo
+        ! In case data is deeper than model
+        hsrc(nz_data) = hsrc(nz_data) + ( zTopOfCell + G%bathyT(CS%col_i(c),CS%col_j(c)) )
+        CS%Ref_val(m)%h(1:nz_data,c) = GV%Z_to_H*hsrc(1:nz_data)
+        CS%Ref_val(m)%p(1:nz_data,c) = tmpT1d(1:nz_data)
+        do k=2,nz_data
+          if (CS%Ref_val(m)%h(k,c) <= 0.001*GV%m_to_H) &
+            ! some confusion here about why the masks are not correct returning from horiz_interp
+            ! reverting to using a minimum thickness criteria
+            CS%Ref_val(m)%p(k,c) = CS%Ref_val(m)%p(k-1,c)
+        enddo
       enddo
       deallocate(sp_val, mask_z, hsrc, tmpT1d)
     enddo
-  else
-    nz_data = CS%nz_data
   endif
 
-  allocate(tmp_val2(nz_data))
+  tmp_val1(:)=0.0;h_col(:)=0.0
   do m=1,CS%fldno
+    nz_data = CS%Ref_val(m)%nz_data
+    allocate(tmp_val2(CS%Ref_val(m)%nz_data))
+    if (CS%id_sp_tendency(m) > 0) then
+      allocate(tmp(G%isd:G%ied,G%jsd:G%jed,nz));tmp(:,:,:) = 0.0
+    endif
     do c=1,CS%num_col
-! c is an index for the next 3 lines but a multiplier for the rest of the loop
-! Therefore we use c as per C code and increment the index where necessary.
+      ! c is an index for the next 3 lines but a multiplier for the rest of the loop
+      ! Therefore we use c as per C code and increment the index where necessary.
       i = CS%col_i(c) ; j = CS%col_j(c)
       damp = dt * CS%Iresttime_col(c)
       I1pdamp = 1.0 / (1.0 + damp)
       tmp_val2(1:nz_data) = CS%Ref_val(m)%p(1:nz_data,c)
       if (CS%time_varying_sponges) then
         call remapping_core_h(CS%remap_cs, nz_data, CS%Ref_val(m)%h(1:nz_data,c), tmp_val2, &
-                              CS%nz, h(i,j,:), tmp_val1, h_neglect, h_neglect_edge)
+                              CS%nz, h_col, tmp_val1, h_neglect, h_neglect_edge)
```

Comment:
h_col needs to be defined as h(i,j,:) before this line, otherwise it is  always 0.

Reply:
Ouch, good spotting @abozec.  I agree, this needs a patch ASAP, and will need re-testing.

Reply:
Agreed, and good catch. I'm not sure why `h_col` was introduced here. The array notation that was replaced looks correct and valid.

Reply:
This does not change  answers for tc4, which is confusing. This oversight applies to both  the time_varying and static sponge targets at L967.

Reply:
This only matter if DEFAULT_2018_ANSWERS is set to False (it is True in all the tc cases) due to a masking issue with the ALE sponge target which rendered this bug inconsequential for tc4.  Hazards of not exercising code in production runs where we actually look at the results.

Reply:
Thanks @abozec and @MJHarrison-GFDL , this fix has been applied.